File: services/admin-web/src/types/clearing.ts
export type ClearingStatus = "PENDING" | "SENT" | "CONFIRMED" | "FAILED";

export interface ClearingBatch {
  id: string;
  merchant_id: string;
  date_from: string;
  date_to: string;
  total_amount: number;
  operations_count: number;
  status: ClearingStatus;
  created_at: string;
  updated_at: string;
}

export interface ClearingBatchOperation {
  id: string;
  batch_id: string;
  operation_id: string;
  amount: number;
}

File: platform/processing-core/app/schemas/clearing.py
from datetime import date, datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class ClearingBatchOperationOut(BaseModel):
    id: UUID
    batch_id: UUID
    operation_id: str
    amount: int

    model_config = ConfigDict(from_attributes=True)


class ClearingBatchOut(BaseModel):
    id: UUID
    merchant_id: str
    date_from: date
    date_to: date
    total_amount: int
    operations_count: int
    status: str
    created_at: datetime
    updated_at: datetime
    operations: Optional[List[ClearingBatchOperationOut]] = None

    model_config = ConfigDict(from_attributes=True)


class BuildBatchRequest(BaseModel):
    date_from: date
    date_to: date
    merchant_id: str


class ClearingBatchAdminOut(BaseModel):
    id: str
    batch_date: date
    merchant_id: str
    currency: str
    total_amount: int
    status: str
    details: Optional[list] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ClearingBatchListResponse(BaseModel):
    items: List[ClearingBatchAdminOut]
    total: int
    limit: int
    offset: int
